internal/reader/home_chain.go

```diff
@@ -114,24 +123,46 @@ func (r *homeChainPoller) poll() {
 }
 
 func (r *homeChainPoller) fetchAndSetConfigs(ctx context.Context) error {
-	var chainConfigInfos []ChainConfigInfo
-	err := r.homeChainReader.GetLatestValue(
-		ctx,
-		consts.ContractNameCCIPConfig,
-		consts.MethodNameGetAllChainConfigs,
-		primitives.Unconfirmed,
-		nil,
-		&chainConfigInfos,
-	)
-	if err != nil {
-		return err
+	var allChainConfigInfos []ChainConfigInfo
+	pageIndex := uint64(0)
+
+	for pageIndex < pageIndexHardLimit {
+		var chainConfigInfos []ChainConfigInfo
+		err := r.homeChainReader.GetLatestValue(
+			ctx,
+			consts.ContractNameCCIPConfig,
+			consts.MethodNameGetAllChainConfigs,
+			primitives.Unconfirmed,
+			map[string]interface{}{
+				"pageIndex": pageIndex,
+				"pageSize":  defaultConfigPageSize,
+			},
+			&chainConfigInfos,
+		)
+		if err != nil {
+			return fmt.Errorf("get config index:%d pagesize:%d: %w", pageIndex, defaultConfigPageSize, err)
+		}
+
+		if len(chainConfigInfos) == 0 {
```

Comment:
suggestion: you can optimize this loop by checking `len(chainConfigInfos) < pageSize` - in which case you can break, because if the number of returned entries is < pageSize there will be no more pages.

Reply:
updayed